lib/sequel/plugins/enum_values.rb

```diff
@@ -68,20 +72,24 @@ def enum_values(field)
 
 				private
 
-				def transform_predicate_methods_to_enum_fields(predicate_methods)
-					case predicate_methods
-					when TrueClass
-						all_enum_fields.keys
-					when FalseClass
-						[]
-					else
-						Array(predicate_methods)
+				def process_enum_values_options
+					process_enum_values_option(:filter_methods) do |field, value|
+						dataset_module { where value, field => value }
+					end
+
+					process_enum_values_option(:predicate_methods) do |field, value|
+						define_method("#{value}?", -> { public_send(field) == value })
 					end
 				end
 
-				def define_predicate_method(field, enum_value)
-					define_method "#{enum_value}?" do
-						public_send(field) == enum_value
+				def process_enum_values_option(key)
+					return unless block_given? && (option = @enum_values_options[key])
+
+					enum_fields = all_enum_fields
+					enum_fields = enum_fields.slice(*Array(option)) unless option == true
+
+					enum_fields.each do |field, options|
+						options[:enum_values].each { |value| yield field, value }
```

Comment:
Please, rename `options` to `db_options`. Or maybe `db_properties`? There are options for plugin and from DB scheme, as I understand. It's confusing.

Reply:
it is "column information". If i rename this, i should rename every "field" to "colum". it is for another PR i guess.

Reply:
i will use "field_schema"

Reply:
Used `field_schema`